shell/components/ResourceList/index.vue

```diff
@@ -150,6 +164,11 @@ export default {
       if (neu && !this.hasFetch) {
         this.$fetchType(this.resource);
       }
+    },
+    resourceQuery(neu, old) { // TODO: RC test
```

Comment:
this is similar to `namespaceFilter`, there's a couple of different use cases that had to cover (see comment). are the same needed use cases needed here?